src/topology/mod.rs

```diff
@@ -727,13 +727,18 @@ async fn handle_errors(
     task: impl Future<Output = Result<TaskOutput, ()>>,
     abort_tx: mpsc::UnboundedSender<()>,
 ) -> Result<TaskOutput, ()> {
-    match AssertUnwindSafe(task).catch_unwind().await {
+    match AssertUnwindSafe(task)
+        .catch_unwind()
+        .await
+        .map_err(|_| ())
+        .and_then(|res| res)
+    {
         Err(_) => {
             error!("An error occurred that vector couldn't handle.");
             let _ = abort_tx.send(());
             Err(())
         }
-        Ok(result) => result,
+        Ok(result) => Ok(result),
```

Comment:
This looks like `.map_err`

Reply:
True, `map_err` could be used here. I always feel weird using `map*` methods purely to produce a side-effect (in this case logging and sending on the abort channel). I wish there was a `tap_err` method or similar that would just give me a reference, but not mutate.

Reply:
I guess someone made a crate for that: https://docs.rs/tap/1.0.0/tap/tap/trait.TapFallible.html